tests/unit/students-serializer.test.js

```diff
@@ -0,0 +1,382 @@
+/* eslint no-unused-expressions: 0 */
+
+const appRoot = require('app-root-path');
+const chai = require('chai');
+const chaiAsPromised = require('chai-as-promised');
+const chaiSubset = require('chai-subset');
+const _ = require('lodash');
+const randomize = require('randomatic');
+const sinon = require('sinon');
+
+const studentsSerializer = appRoot.require('api/v1/serializers/students-serializer');
+const testData = appRoot.require('tests/unit/test-data');
+const { openapi } = appRoot.require('utils/load-openapi');
+
+chai.should();
+chai.use(chaiAsPromised);
+chai.use(chaiSubset);
+const { expect } = chai;
+
+describe('Test students-serializer', () => {
+  const { fakeId, fakeBaseUrl } = testData;
+  const resourceSubsetSchema = (resourceType, resourceAttributes) => {
+    const schema = {
+      links: {
+        self: `${fakeBaseUrl}/${resourceType}`,
+      },
+      data: {
+        id: fakeId,
+        type: resourceType,
+        links: { self: null },
+      },
+    };
+    if (resourceAttributes) {
+      schema.data.attributes = resourceAttributes;
+    }
+    return schema;
+  };
+
+  /**
+   * @summary Helper function for lite-testing single resource
+   * @function
+   * @param {object} serializedResource serialized resource
+   * @param {string} resourceType resource type
+   * @param {string} nestedProps field name of the nested properties
+   */
+  const testSingleResource = (serializedResource, resourceType, nestedProps) => {
+    expect(serializedResource).to.containSubset(resourceSubsetSchema(resourceType));
+
+    if (nestedProps) {
+      expect(serializedResource).to.have.nested.property(`data.attributes.${nestedProps}`);
+    }
+  };
+
+  /**
+   * @summary Helper function for lite-testing multiple resources
+   * @function
+   * @param {Object} serializedResources serialized resources
+   * @returns {Object} data object from serialized resources for further use
+   */
+  const testMultipleResources = (serializedResources) => {
+    const serializedResourcesData = serializedResources.data;
+    expect(serializedResources).to.have.keys('data', 'links');
+    expect(serializedResourcesData).to.be.an('array');
+
+    return serializedResourcesData;
+  };
+
+  /**
+   * @summary Helper function to get definition from openapi specification
+   * @function
+   * @param {string} definition the name of definition
+   * @param {Object} nestedOption nested option
+   * @param {boolean} dataItem a boolean value which represents whether it's a data item
+   * @param {string} dataField data field name
```

Comment:
I think these should be named `nestedOption.dataItem` and `nestedOption.dataField` http://usejsdoc.org/tags-param.html#parameters-with-properties

Reply:
Ah, right. Fixed.